File: python/GeometryBasicsDistancebetweenpointsin3D.py
# Kata: Geometry Basics: Distance between points in 3D
# Level: 7kyu
# Solved: 2025-08-10

class Point:
    def __init__(self, x: float, y: float , z: float):
        self.x = x
        self.y = y
        self.z = z

def distance_between_points(a: Point, b: Point) -> float:
    return ((a.x - b.x) ** 2 + (a.y- b.y) ** 2 + (a.z - b.z) ** 2 ) ** 0.5

File: js/GeometryBasicsDistancebetweenpointsin3D.js
// Kata: Geometry Basics: Distance between points in 3D
// Level: 7kyu
// Solved: 2025-08-10

function distanceBetweenPoints(a, b) {
    return Math.sqrt((Math.pow(b.x - a.x, 2)) + (Math.pow(b.y - a.y, 2)) + (Math.pow(b.z - a.z, 2)));
}

class Point {
    constructor(x = 0, y = 0, z = 0) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}